tests/yast2_gui/bootloader/bootcode_options.pm

```diff
@@ -0,0 +1,87 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+#
+# Summary: Open bootloader gui, verify default setting in ui, change
+# some settings and verify that they have been applied.
+#
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base "y2_module_guitest";
+use strict;
+use warnings;
+use testapi;
+use scheduler 'get_test_suite_data';
+use YaST::Module;
+use cfg_files_utils;
+use x11utils 'start_root_shell_in_xterm';
+use y2lan_restart_common 'close_xterm';
+
+my $bootloader;
+my $test_data;
+
+sub verify_current_options {
+    my ($args) = @_;
+    YaST::Module::run_actions {
+        my %current_settings = $bootloader->get_current_settings();
+        compare_settings({expected => $test_data->{bootcode_options}, current => \%current_settings});
+        $bootloader->accept_changes();
```

Comment:
close() fails if we press cancel, the return code is then not zero.

Reply:
if you don't mind to check that behaviour with YaST developers, is it documented what recode we got?, perhaps there is a different between failure and cancel, it would good to know for the future. Anyway we don't need to Cancel, more simple workflow is to check first by command line what is expected in the first step and then go to the module for the first time, check what is expected in the UI and apply the change and click OK. To click Cancel was just one way to achieve to check both UI and console before touching anything.

Reply:
I reported https://bugzilla.suse.com/show_bug.cgi?id=1186132 after talking with Yast team, let's see if they confirm, but I'm pretty sure they will. I was a bit puzzled by the error in openqa though. Te module returns 16, but the error in openqa shows "yast2-bootloader-status-4".
Anyway this is pretty annoying, as Yast returns 16 when it fails as well.

Reply:
fixed using a different flow.